### ISSUE TITLE
Nottingham City Council - Error

### ISSUE THREAD

**Issue by newelley** [mampfes/hacs_waste_collection_schedule#1117]
This is my first Issue submission on Github, I apologise in advance if I do this incorrectly, Please advise if so and I'll endeavor to change/add any relevant information :)

This Council used to work fine in my home assistant setup, but now, where it used to say "next bin collection in x days" it just has the "unknown" text.

I checked the logs and have the following error:
```

This error originated from a custom integration.

Logger: waste_collection_schedule.source_shell
Source: custom_components/waste_collection_schedule/waste_collection_schedule/source_shell.py:136
Integration: waste_collection_schedule (documentation)
First occurred: 10:52:00 (1 occurrences)
Last logged: 10:52:00

fetch failed for source Nottingham City Council: Traceback (most recent call last): File "/config/custom_components/waste_collection_schedule/waste_collection_schedule/source_shell.py", line 134, in fetch entries = self._source.fetch() File "/config/custom_components/waste_collection_schedule/waste_collection_schedule/source/nottingham_city_gov_uk.py", line 46, in fetch data = json.loads(r.text) File "/usr/local/lib/python3.10/json/__init__.py", line 346, in loads return _default_decoder.decode(s) File "/usr/local/lib/python3.10/json/decoder.py", line 337, in decode obj, end = self.raw_decode(s, idx=_w(s, 0).end()) File "/usr/local/lib/python3.10/json/decoder.py", line 355, in raw_decode raise JSONDecodeError("Expecting value", s, err.value) from None json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
```

I would very much like to get this working again.  does anyone have any suggestions or know how to fix this?  Again, I am happy to supply information which may be useful to debug.

**Comment by newelley** [comment]
Ok, a little more information, if it is useful...

I have looked at the URLs that are requested by nottingham_city_gov_uk.py and the current URL supplied to my browser after follwing the https://www.nottinghamcity.gov.uk/binreminders link to discover my uprn.  It appears there may be a mismatch:
```
URL requested from nottingham_city_gov_uk.py = https://geoserver.nottinghamcity.gov.uk/myproperty/handler/proxy.ashx?http://geoserver.nottinghamcity.gov.uk/wcf/BinCollection.svc/livebin/{self._uprn}
```
This returns a 403 - Forbidden error

```
URL given by https://www.nottinghamcity.gov.uk/binreminders = https://collectiveview.bartec-systems.com/R152/CollectiveCalendar.aspx?Token=ppfKTw2tn4zo37i8eWv%2BR1iClm8Kka%2B%2FKnZoCpgt%2FNY%2BNTNGovD3Vol%2FImztoADp0AIPHN775DzZmShhw0PVEqMrugatxxkfdGmTxrJCU8uVt7LIhE84hA%3D%3D&UPRN=100031553830&Width=770&Align=center&Theme=flick#
```
using the test case uprn in both examples.

Is this just as simple as substituting the working URL in the .py file?

**Comment by dt215git** [comment]
Test cases still seem to work:

```bash
Testing source nottingham_city_gov_uk ...
  found 3 entries for Douglas Rd, Nottingham NG7 1NW
    2023-08-30 : General [mdi:trash-can]
    2023-09-06 : Recycling [mdi:recycle]
    2023-09-06 : Garden [mdi:leaf]
  found 3 entries for Harlaxton Drive, Nottingham, NG7 1JE
    2023-08-31 : Recycling [mdi:recycle]
    2023-08-31 : Garden [mdi:leaf]
    2023-09-07 : General [mdi:trash-can]
```
And with a browser I get a valid responses from the geoserver url when substituting in a uprn.
![image](https://github.com/mampfes/hacs_waste_collection_schedule/assets/86194065/f33d4fe2-51e3-4c14-9002-6e491e1783b1)

Any chance something else has changed on your setup?

**Comment by newelley** [comment]
The fix issued in version 1.42 resolved this problem for me.  It appears (with my limited knowledge...) that there was some mismatch with the API URL which works now corrected.  Should I close this issue now?
